src/main.rs

```diff
@@ -157,3 +244,47 @@ fn parse_line(line: &str) -> anyhow::Result<Option<DVec3>> {
         _ => Ok(None),
     }
 }
+
+fn histogram(
+    positions: &Vec<DVec3>,
+    r_variable: fn(&DVec3) -> f64,
+    (avg, std_dev): (DVec3, DVec3),
+) -> (Vec<Vec<DVec3>>, Vec<(f64, f64)>) {
+    let mut position_set: Vec<DVec3> = positions.clone();
+
+    let cutoff: i32 = 3; // measured in standard deviations
+    let div: i32 = 6;
+    let mut range = (-(cutoff * div)..(cutoff * div))
+        .into_iter()
+        .map(|i| (i as f64) / (div as f64) * r_variable(&std_dev) + r_variable(&avg))
+        .peekable();
+    position_set.sort_by(|a, b| {
+        r_variable(a)
+            .partial_cmp(&r_variable(b))
+            .expect("Histogram: Incomparable values")
+    });
+
+    let division_values = range.clone().zip(range.clone().skip(1)).collect::<Vec<_>>();
+    let mut histogram: Vec<Vec<DVec3>> = Vec::new();
+    let mut division_aux: Vec<DVec3> = Vec::new();
+
+    for pos in position_set {
+        while let Some(val) = range.peek() {
+            if r_variable(&pos) < *val {
+                division_aux.push(pos);
+                break;
+            } else {
+                histogram.push(division_aux.clone());
+                division_aux.clear();
+                range.next();
+            }
+        }
+    }
+
+    while range.next().is_some() {
+        histogram.push(division_aux.clone());
+        division_aux.clear();
+    }
```

Comment:
This still makes no sense to me. This will only make sense in a single iteration. After that, the vector to be pushed will empty itself.

Reply:
Okay: This should probably be an if, not a while. Iterating here makes no sense to me.